### PR TITLE
FB Pixel Event ID details 

### PR DESCRIPTION
**Pull request by ishachopra0401**
## Description of the change

Updated event ID details 

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: destinations/advertising/fb-pixel.mdx [hunk 1]
```diff
@@ -340,7 +340,9 @@ Facebook's \(and RudderStack's\) recommended form of deduplication is to leverag
   - The `event_id` must be unique to that specific event and the same for both of the events coming from the Pixel and Conversion API
   - For example: the `event_id` could be the purchase order number etc.
 
-To set the `event_id`, simply include the `event_id` key-value pair either in the `context.traits` object or the `properties` object. The `event_name` will be whatever the name of the event is.
+To set the `event_id`, you can set a unique identifier or a key-value pair in the `context.traits`, `properties`, or `traits` object. It works for cloud as well as device mode. If you don't set the `event_id`, RudderStack will use the `messageId` to map to Facebook's `eventId`.
```
Comment (utsabc) [inline destinations/advertising/fb-pixel.mdx:343]: it should be set as `traits.event_id: unique identifier` for context.traits and `properties` as well

Comment (ishachopra0401) [reply destinations/advertising/fb-pixel.mdx]: @utsabc Please have a look now

File: destinations/advertising/fb-pixel.mdx [hunk 2]
```diff
@@ -340,7 +340,9 @@ Facebook's \(and RudderStack's\) recommended form of deduplication is to leverag
   - The `event_id` must be unique to that specific event and the same for both of the events coming from the Pixel and Conversion API
   - For example: the `event_id` could be the purchase order number etc.
 
-To set the `event_id`, simply include the `event_id` key-value pair either in the `context.traits` object or the `properties` object. The `event_name` will be whatever the name of the event is.
+You can set the `event_id` as `traits.event_id: unique identifier` in the `context.traits` or `properties` object. It works for cloud as well as device mode. If you don't set the `event_id`, RudderStack will use the `messageId` to map to Facebook's `eventId`.
```
Comment (utsabc) [inline destinations/advertising/fb-pixel.mdx:343]:
```suggestion
You can set the `event_id` in `traits` , `context.traits` or `properties` object as an unique identifier. It works for cloud as well as device mode. If you don't set the `event_id`, RudderStack will use the `messageId` to map to Facebook's `eventId`.
```

File: destinations/advertising/fb-pixel.mdx [hunk 3]
```diff
@@ -340,7 +340,9 @@ Facebook's \(and RudderStack's\) recommended form of deduplication is to leverag
   - The `event_id` must be unique to that specific event and the same for both of the events coming from the Pixel and Conversion API
   - For example: the `event_id` could be the purchase order number etc.
 
-To set the `event_id`, simply include the `event_id` key-value pair either in the `context.traits` object or the `properties` object. The `event_name` will be whatever the name of the event is.
+You can set the `event_id` in `traits` , `context.traits` or `properties` object as an unique identifier. It works for cloud as well as device mode. If you don't set the `event_id`, RudderStack will use the `messageId` to map to Facebook's `eventId`.
```
Comment (ameypv-rudder) [inline destinations/advertising/fb-pixel.mdx:343]:
```suggestion
You can set the `event_id` in `traits` , `context.traits` or `properties` object as a unique identifier. This works for the [cloud mode](https://www.rudderstack.com/docs/connections/rudderstack-connection-modes/#cloud-mode) as well as [device mode](https://www.rudderstack.com/docs/connections/rudderstack-connection-modes/#device-mode). 

<div class="infoBlock">

If you don't set the <code class="inline-code">event_id</code>, RudderStack will use <code class="inline-code">messageId</code> to map to Facebook's <code class="inline-code">eventId</code>.
</div>
```

Comment (ameypv-rudder) [reply destinations/advertising/fb-pixel.mdx:343]: Made minor structural changes and added references for better context/understanding.

**Review by ameypv-rudder — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**